pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -963,7 +973,106 @@ func (a *Actuator) remediateIfNeeded(ctx context.Context, machine *machinev1.Mac
 		return &clustererror.RequeueAfterError{RequeueAfter: time.Second * 5}
 	}
 
-	// node is now available again
+	// node is now available again - restore annotations and labels
+	if _, annotationsBackupExists := machine.Annotations[nodeAnnotationsBackupAnnotation]; annotationsBackupExists {
+		return a.restoreAnnotationsAndLabels(node, machine, ctx)
+	}
+
+	//remediation is done
 	log.Printf("Node %s is available, remediation of Machine %s complete", node.Name, machine.Name)
 	return a.deleteRemediationAnnotations(ctx, machine)
 }
+
+// storeAnnotationsAndLabels copies node's annotations and labels and stores them on machine's annotations
+func (a *Actuator) storeAnnotationsAndLabels(node *corev1.Node, machine *machinev1.Machine, ctx context.Context) error {
+	marshaledAnnotations, err := marshal(node.Annotations)
+	if err != nil {
+		log.Printf("Failes to marshal node %s annotations assoicated with Machine %s: %s",
```

Comment:
s/Failes/Fails/
s/assoicated/associated/

Reply:
> s/Failes/Fails/

I believe it should be Failed, right?

> s/assoicated/associated/

indeed. thanks.